packages/vim.rb

```diff
@@ -3,79 +3,106 @@
-    configdestdir = "#{CREW_DEST_PREFIX}/.config/vim"
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+
     system "compressdoc --gzip -9 #{CREW_DEST_PREFIX}/share/man/man1"
-    FileUtils.mkdir_p "#{configdestdir}"
-    system "cp -r #{CREW_DEST_PREFIX}/share/vim/vim81/* #{configdestdir}"
-    FileUtils.mkdir_p "#{CREW_DEST_HOME}"
-    FileUtils.mkdir_p "#{configdir}" unless Dir.exists? "#{configdir}"
-    system "ln -s #{configdir} #{CREW_DEST_HOME}/.vim"
-    system "ln -s #{configdir}/vimrc #{CREW_DEST_HOME}/.vimrc"
-    system "touch #{configdir}/vimrc #{configdestdir}/vimrc"
+
+    # remove desktop and icon files for the terminal package
+    system "rm", "-r", "#{CREW_DEST_PREFIX}/share/applications"
+    system "rm", "-r", "#{CREW_DEST_PREFIX}/share/icons"
+
+    # add some defaults simulating XDG support
+    FileUtils.mkdir_p("#{CREW_DEST_PREFIX}/share/vim/vimfiles")
+    File.write("#{CREW_DEST_PREFIX}/share/vim/vimfiles/chromebrew.vim", <<~EOF)
+      " Global vimrc - setting some sane defaults
+      "
+      " DO NOT EDIT THIS FILE. IT'S OVERWRITTEN UPON UPGRADES.
+      "
+      " Use #{CREW_PREFIX}/etc/vimrc for system-wide and ~/.vimrc for personal
+      " configuration.
+
+      " Use Vim defaults instead of 100% vi compatibility
+      " Avoid side-effects when nocompatible has already been set.
+      if &compatible
+        set nocompatible
+      endif
+
+      set backspace=indent,eol,start
+      set ruler
+      set suffixes+=.aux,.bbl,.blg,.brf,.cb,.dvi,.idx,.ilg,.ind,.inx,.jpg,.log,.out,.png,.toc
+      set suffixes-=.h
+      set suffixes-=.obj
+
+      " Move temporary files to a secure location to protect against CVE-2017-1000382
+      if exists('$XDG_CACHE_HOME')
+        let &g:directory=$XDG_CACHE_HOME
+      else
+        let &g:directory=$HOME . '/.cache'
+      endif
+      let &g:undodir=&g:directory . '/vim/undo//'
+      let &g:backupdir=&g:directory . '/vim/backup//'
+      let &g:directory.='/vim/swap//'
+      " Create directories if they doesn't exist
+      if ! isdirectory(expand(&g:directory))
+        silent! call mkdir(expand(&g:directory), 'p', 0700)
+      endif
+      if ! isdirectory(expand(&g:backupdir))
+        silent! call mkdir(expand(&g:backupdir), 'p', 0700)
+      endif
+      if ! isdirectory(expand(&g:undodir))
+        silent! call mkdir(expand(&g:undodir), 'p', 0700)
+      endif
+    EOF
   end
 
   def self.postinstall
-    configdir = "#{CREW_PREFIX}/.config/vim"
-    system "cp #{CREW_PREFIX}/share/vim/vim81/vimrc_example.vim #{configdir}/vimrc" unless File.size? "#{configdir}/vimrc"
-    puts
-    puts "To finish the installation, execute the following:".lightblue
-    puts "echo 'export VIMRUNTIME=$HOME/.vim' >> ~/.bashrc && . ~/.bashrc".lightblue
-    puts
-    puts "The config files are located in $HOME/.vim".lightblue
-    puts
+    vimrc = "#{CREW_PREFIX}/etc/vimrc"
+    # keep user changes by writing to a new file
+    vimrc += ".new" if File.exists?(vimrc)
```

Comment:
Might want to create a random extension since `vimrc.new` may already exist from a previous upgrade.

Reply:
That's fine, the user is expected to diff that file again the `vimrc` in place and keep what they want, they're free to delete the `.new` afterwards. We just don't want to replace the potentially user-modified `vimrc` file. (which is providing the opt-out for the opiniated config above)

Reply:
Just keep in mind that if they forget about the `.new` file (or if they never knew in the first place), any subsequent installs will overwrite the "saved" file.

Reply:
Why would overwriting the `.new` file be an issue? Its only purpose is to allow users to basically manually install changes from an upgrade. It is actually by-design that the "new" file be always overwritten by the package manager. This allows us to propose changes without imposing them.